src/libraries/System.Text.Json/tests/Serialization/InvalidTypeTests.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class InvalidTypeTests_Span : InvalidTypeTests
+    {
+        public InvalidTypeTests_Span() : base(SerializationWrapper.SpanSerializer) { }
+    }
+
+    public class InvalidTypeTests_String : InvalidTypeTests
+    {
+        public InvalidTypeTests_String() : base(SerializationWrapper.StringSerializer) { }
+    }
+
+    public class InvalidTypeTests_Stream : InvalidTypeTests
+    {
+        public InvalidTypeTests_Stream() : base(SerializationWrapper.StreamSerializer) { }
+    }
+
+    public class InvalidTypeTests_StreamWithSmallBuffer : InvalidTypeTests
+    {
+        public InvalidTypeTests_StreamWithSmallBuffer() : base(SerializationWrapper.StreamSerializerWithSmallBuffer) { }
+    }
+
+    public class InvalidTypeTests_Writer : InvalidTypeTests
+    {
+        public InvalidTypeTests_Writer() : base(SerializationWrapper.WriterSerializer) { }
+    }
+
+    public abstract class InvalidTypeTests
+    {
+        private SerializationWrapper Serializer { get; }
+
+        public InvalidTypeTests(SerializationWrapper serializer)
+        {
+            Serializer = serializer;
+        }
+
+        [Theory]
+        [MemberData(nameof(OpenGenericTypes))]
+        [MemberData(nameof(RefStructTypes))]
+        [MemberData(nameof(PointerTypes))]
+        public void DeserializeInvalidType(Type type)
```

Comment:
Can you add a test case for `Memory<T>` as well please (for both serialize and deserialize)? Also, just for completeness, add a test for `Utf8JsonReader` (a ref-struct type containing another ref-struct as a field).

Reply:
A test for `Memory<T>` is included in https://github.com/dotnet/runtime/blob/d0b6b3f52b94b81e60784c831c9a9c8a16ed1e2c/src/libraries/System.Text.Json/tests/Serialization/InvalidTypeTests.cs#L56.